spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ArrowVectorAccessors.java

```diff
@@ -0,0 +1,495 @@
+      Binary binary = parquetDictionary.decodeToBinary(dictionaryVector.get(rowId));
+      return UTF8String.fromBytes(binary.getBytesUnsafe());
+    }
+  }
+
+  private static class FixedSizeBinaryAccessor extends ArrowVectorAccessor {
+
+    private final FixedSizeBinaryVector vector;
+
+    FixedSizeBinaryAccessor(FixedSizeBinaryVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final byte[] getBinary(int rowId) {
+      return vector.getObject(rowId);
+    }
+  }
+
+  private static class BinaryAccessor extends ArrowVectorAccessor {
+
+    private final VarBinaryVector vector;
+
+    BinaryAccessor(VarBinaryVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final byte[] getBinary(int rowId) {
+      return vector.getObject(rowId);
+    }
+  }
+
+  private static class DictionaryBinaryAccessor extends DictionaryArrowVectorAccessor {
+
+    DictionaryBinaryAccessor(IntVector vector, Dictionary dictionary) {
+      super(vector, dictionary);
+    }
+
+    @Override
+    final byte[] getBinary(int rowId) {
+      Binary binary = parquetDictionary.decodeToBinary(dictionaryVector.get(rowId));
+      return binary.getBytesUnsafe();
+    }
+  }
+
+  private static class DateAccessor extends ArrowVectorAccessor {
+
+    private final DateDayVector vector;
+
+    DateAccessor(DateDayVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final int getInt(int rowId) {
+      return vector.get(rowId);
+    }
+  }
+
+  private static class TimestampAccessor extends ArrowVectorAccessor {
+
+    private final TimeStampMicroTZVector vector;
+
+    TimestampAccessor(TimeStampMicroTZVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final long getLong(int rowId) {
+      return vector.get(rowId);
+    }
+  }
+
+  private static class ArrayAccessor extends ArrowVectorAccessor {
+
+    private final ListVector vector;
+    private final ArrowColumnVector arrayData;
+
+    ArrayAccessor(ListVector vector) {
+      super(vector);
+      this.vector = vector;
+      this.arrayData = new ArrowColumnVector(vector.getDataVector());
+    }
+
+    @Override
+    final ColumnarArray getArray(int rowId) {
+      ArrowBuf offsets = vector.getOffsetBuffer();
+      int index = rowId * ListVector.OFFSET_WIDTH;
+      int start = offsets.getInt(index);
+      int end = offsets.getInt(index + ListVector.OFFSET_WIDTH);
+      return new ColumnarArray(arrayData, start, end - start);
+    }
+  }
+
+  /**
+   * Use {@link IcebergArrowColumnVector#getChild(int)} to get hold of the {@link ArrowColumnVector} vectors holding the
+   * struct values.
+   */
+  private static class StructAccessor extends ArrowVectorAccessor {
+    StructAccessor(StructVector structVector) {
+      super(structVector);
+      childColumns = new ArrowColumnVector[structVector.size()];
```

Comment:
This is the only place that uses `childColumns` and it seems really strange to only set them in a sub-class, even though they are available in the constructor.

Could you make the superclass `childColumns` a `private final` field instead, and pass the child columns through a second constructor? Like this:

```java
    StructAccessor(StructVector structVector) {
      super(structVector,
          IntStream.range(0, structVector.size())
              .mapToObj(structVector::getVectorById)
              .map(ArrowColumnVector::new)
              .toArray(ArrowColumnVector[]::new));
    }
```